src/join.sol

```diff
@@ -427,3 +427,113 @@ contract AuthGemJoin is LibNote {
         require(gem.transfer(usr, wad), "AuthGemJoin/failed-transfer");
     }
 }
+
+interface GemLike7 {
+    // matches ERC20 spec
+    function decimals() external view returns (uint);
+    // matches ERC20 spec
+    function transfer(address,uint) external;
+    // matches ERC20 spec
+    function transferFrom(address,address,uint) external;
+    // USDT uses balanceOf() constant
+    function balanceOf(address) external view returns (uint);
+    // USDT uses allowance() constant
+    function allowance(address,address) external view returns (uint);
+    // doesn't match ERC20 spec
+    function upgradedAddress() external view returns (address);
+    // doesn't match ERC20 spec
+    function deprecated() external view returns (bool);
+    // doesn't match ERC20 spec
+    function setImplementation(address,uint) external;
+}
+
+contract GemJoin7 is LibNote {
+    mapping (address => uint) public wards;
+    function rely(address usr) external note auth { wards[usr] = 1; }
+    function deny(address usr) external note auth { wards[usr] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+
+    VatLike  public vat;
+    bytes32  public ilk;
+    GemLike7 public gem;
+    uint     public dec;
+    uint     public live; // Access flag
+
+    mapping (address => uint256) public implementations;
+
+    constructor(address vat_, bytes32 ilk_, address gem_) public {
+        gem = GemLike7(gem_);
+        dec = gem.decimals();
+        require(dec < 18, "GemJoin7/decimals-18-or-higher");
+        wards[msg.sender] = 1;
+        live = 1;
+        vat = VatLike(vat_);
+        ilk = ilk_;
+        setImplementation(address(gem), 1);
```

Comment:
This adds the `gem` address to the implementations array (currently `
0xdac17f958d2ee523a2206206994597c13d831ec7`), but it will never get checked.

You'd want to do `setImplementation(gem.upgradedAddress(), 1);` here. Even though it's currently `0x000...` we'd still want it to be approved.

Reply:
resolved https://github.com/makerdao/dss-deploy/pull/53/commits/53c06b3097d6f8513b6dd7447df44b029fbb3105